app/Theme/Plugins.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Dxw\MyTheme\Theme;
+
+use Dxw\Iguana\Registerable;
+
+class Plugins implements Registerable
+{
+    protected $required;
+    protected $abspath;
+
+    public function __construct(array $required = [], $abspath = null)
+    {
+        $this->required = $required;
+        $this->abspath = $abspath ?? ABSPATH;
+    }
+
+    public function register()
+    {
+        add_action('after_switch_theme', [$this, 'checkDependencies']);
```

Comment:
I suppose the theme could get activated programatically via `update_option('stylesheet, $theme_name);` However I can't think of a situation where this might happen.

Unfortunately there's no equivalent of the plugin action `register_activation_hook` for themes. The action I've used is the closest, the docs say:
```
Fires on the first WP load after a theme switch if the old theme still exists.
```